controller/appcontroller.go

```diff
@@ -2360,6 +2363,34 @@ func (ctrl *ApplicationController) newApplicationInformerAndLister() (cache.Shar
 	if err != nil {
 		return nil, nil
 	}
+
+	// Periodically updates ObservedAt time in Application health status to indicate freshness
+	_, err = informer.AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(old, new interface{}) {
+				if !ctrl.canProcessApp(new) {
+					return
+				}
+				oldApp, oldOK := old.(*appv1.Application)
+				newApp, newOK := new.(*appv1.Application)
+				if oldOK && newOK {
+					// To prevent a status update hot loop, enqueue only events generated by resyncs
+					// and ignore those triggered by application updates, which are handled by another event handler.
+					if oldApp.ResourceVersion != newApp.ResourceVersion {
+						return
+					}
+
+					// We are interested only in refreshing the status, not performing a resource-level comparison.
+					ctrl.requestAppRefresh(newApp.QualifiedName(), ComparisonWithNothing.Pointer(), nil)
```

Comment:
Hm, I have concerns about triggering a refresh every 30 seconds for each app, this may add up.